tree/ntuple/v7/inc/ROOT/RFieldVisitor.hxx

```diff
@@ -0,0 +1,77 @@
+/// \file ROOT/RFieldVisitor.hxx
+/// \ingroup NTuple ROOT7
+/// \author Simon Leisibach <simon.satoshi.rene.leisibach@cern.ch>
+/// \date 2019-06-11
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT7_RFieldVisitor
+#define ROOT7_RFieldVisitor
+
+#include <iostream>
+#include <sstream>
+#include <string>
+
+namespace ROOT {
+namespace Experimental {
+class RFieldRoot;
+namespace Detail {
+class RFieldBase;
+}
+
+// clang-format off
+/**
+\class ROOT::Experimental::RNTupleVisitor
+\ingroup NTuple
+\brief Pure virtual parentclass for classes implementing the visitor design pattern.
+     
+Currently only has a virtual visitField() function, used by the RPrintVisitor class.
+*/
+// clang-format on
+class RNTupleVisitor {
+public:
+   virtual void visitField(const ROOT::Experimental::Detail::RFieldBase &fField) = 0;
+};
+
+// clang-format off
+/**
+\class ROOT::Experimental::RPrintVisitor
+\ingroup NTuple
+\brief Contains settings for printing and prints a summary of an RField instance.
+     
+ Instances of this class are currently only invoked by RNTupleReader::Print() -> RFieldBase::AcceptVisitor()
```

Comment:
That doesn't match my expectation. I would gave expected that I can invoke `RNTuple::VisitFields(visitor)` which calls `visitor.VisitField()` for each of the fields in the ntuple? Maybe we should discuss that verbally :-)